### ISSUE TITLE
Is standalone DocBlocks using #ref not supported in swagger 4.x?

### ISSUE THREAD

**Issue by nirajfu3e** [zircote/swagger-php#1398]
We have a code that looks like this:

```php
class AccountController extends Controller 
{
    /**
    * @OA\Get (
    *     path="/api/v1/accounts/{id}",
    *     @OA\Response(
    *     response="200",
    *     description="AccountResource",
    *        @OA\JsonContent(ref="#/components/schemas/AccountResource"),
    *     ),
    * )
    */
    public function account (int $id): AccountResource
    {
           $account = $this->accountRepository->find($id);
           return AccountResource::make($account);
    }
}
```
The referenced `AccountResource` has the following code:
```php
/**
 * @OA\Schema(
 *  description="Account Resource",
 *  type="object",
 *  title="Account Resource",
 *      @OA\Property(property="id", type="integer", example="123"),
 *      @OA\Property(property="name", type="string", example="John Doe"),
 * )
 */
class AccountResource extends JsonResource
{
}
```
We recently upgraded the package from 3.x to 4.x and all of a sudden we are seeing a ton of these errors wherever we are using the `ref`.
```bash
Warning: $ref "#/components/schemas/AccountResource" not found for @OA\JsonContent()
```

I found an answer in [stackoverflow ](https://stackoverflow.com/a/71192376/2165516) which seems to suggest using `ref` is not supported anymore in 4.x? Is that correct or is there a way to fix this without downgrading to 3.x? 

Or is there a way in 4.x where we can reuse the `AccountResource` using PHP attributes instead of using annotations?

**Comment by temp** [comment]
Same problem here, but with Attributes.
Seems like the Schema is complete ignored, it is not generated inside the doc.json.

**Comment by DerManoMann** [comment]
That docblock looks ok to me. 
Could someone please create a single file reproducer? Still not clear what exactly the issue is here.

**Comment by DerManoMann** [comment]
Maybe it is the `standalone docblock` thing...?

Taking your code sample the following does work. However, it has the `account` method moved into the controller class..
The main issue upgrading from 3 to 4 seems to be that people (me included) used to create standalone docblock annotations without associated code. swagger-php v4 does use reflection in a few places, so without a class, method, etc. to use reflection on it will not 'see' docblocks.
This brings the use of docblocks en par with attributes which also cannot exist on their own.

```php
<?php

use OpenApi\Annotations as OA;

/**
 * @OA\Schema(
 *  description="Account Resource",
 *  type="object",
 *  title="Account Resource",
 *      @OA\Property(property="id", type="integer", example="123"),
 *      @OA\Property(property="name", type="string", example="John Doe")
 * )
 */
class AccountResource
{
}

class AccountController
{
    /**
     * @OA\Get (
     *     path="/api/v1/accounts/{id}",
     *     @OA\Response(
     *        response="200",
     *        description="AccountResource",
     *        @OA\JsonContent(ref="#/components/schemas/AccountResource")
     *     )
     * )
     */
    public function account(int $id): AccountResource
    {
        return new AccountResource();
    }
}
```

```sh
   ./bin/openapi -b docblock_ref.php docblock_ref.php
```


This results in:

```yaml
openapi: 3.0.0
paths:
  '/api/v1/accounts/{id}':
    get:
      operationId: 842d94638eba5f319a79fa6bf9ee8823
      responses:
        '200':
          description: AccountResource
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/AccountResource'
components:
  schemas:
    AccountResource:
      title: 'Account Resource'
      description: 'Account Resource'
      properties:
        id:
          type: integer
          example: '123'
        name:
          type: string
          example: 'John Doe'
      type: object
```

**Comment by nirajfu3e** [comment]
It would indeed be the standalone thing as the `AccountResource` class existed on its own since it is re-used in many controllers. With the suggested fix of moving the `AccountResource` class within the controller itself, I guess that would mean having to sacrifice re-usability completely which I feel like is not worth the upgrade to 4.x.

**Comment by DerManoMann** [comment]
That is a misunderstanding. I just moved everything into a single file for simplicity. Splitting the example into two files will not make a difference.

**Comment by nirajfu3e** [comment]
I am so sorry @DerManoMann as I just realised my mistake in the initial question as I did not mean to put the `account` method outside of the controller class. I was checking your suggested fix and my own code and was confused what was different and realised the mistake I made in the initial question.

Unfortunately, it has the exact same error even when I copy pasted your DocBlocks for the `AccountController` and `AccountResource`.
